public/scripts/workflow.js

```diff
@@ -1,41 +1,94 @@
 function parseRepositoryURL(repoURL) {
-  if (repoURL.startsWith("git@") || repoURL.startsWith("ssh://")) {
+  repoURL = repoURL.trim().replace(/\.git$/i, "");
+
+  // SSH and git protocol (e.g., git@github.com:owner/repo or git://host/owner/repo)
+  const m = repoURL.match(
+    /^(git@|ssh:\/\/git@|git:\/\/)([^:/]+)[:/]((?:[^/]+\/)+[^/]+)$/i,
+  );
+  if (m) {
+    const host = m[2];
+    const parts = m[3].split("/").filter(Boolean);
+    // GitHub Enterprise SSH URL: git@github.com:enterprises/enterprise/owner/repo
+    if (
+      host.toLowerCase() === "github.com" &&
+      parts[0] === "enterprises" &&
+      parts.length >= 4
+    ) {
+      return {
+        origin: `https://github.com`,
+        enterprise: parts[1],
+        owner: parts[2],
+        repo: parts[3],
+      };
+    } else if (parts.length === 2) {
+      // Standard SSH URL: git@host:owner/repo
+      return {
+        origin: `https://${host}`,
+        owner: parts[0],
+        repo: parts[1],
+      };
+    }
+    // Unrecognized SSH format
     return null;
   }
-  if (repoURL.endsWith(".git")) {
-    repoURL = repoURL.slice(0, -4);
+
+  // Shorthand and domain/owner/repo (e.g., owner/repo or host/owner/repo)
+  if (!/^\w+:\/\//.test(repoURL)) {
+    const parts = repoURL.split("/").filter(Boolean);
+    if (parts.length === 2 && parts[0].includes(".")) {
+      // host/owner (missing repo)
+      return {
+        origin: `https://${parts[0]}`,
+        owner: parts[1],
+        repo: parts[2], // will be undefined, so will be caught below
+      };
```

Comment:
The logic looks correct, but I think the comment is off. This looks like the github.com/host/owner case

Reply:
Oh, nevermind. I see it's `parts.length === 2`. I am a little confused about parsing and returning for this case. This is to handle false positive matches where they left off the repo? If we don't include this, we'll accidentally match?